doc/types/Hasher.qbk

```diff
@@ -1,56 +0,0 @@
-[/
-    Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
-
-    Distributed under the Boost Software License, Version 1.0. (See accompanying
-    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
-]
-
-[section:Hasher Hasher]
-
-A [@Hasher] implements a hash algorithm. This is used to compute the small
-digests NuDB needs to effectively implement a hash table. NuDB provides
-the default implementation [link nudb.ref.nudb__xxhasher xxhasher], which is
-suitable for most use cases. For advanced applications, a user supplied
-hash function may be supplied which must meet these requirements.
-
-In the table below:
-
-* `X` denotes a hasher class
-* `a` denotes a value of type `X const`
-* `s` denotes a value of type `std::uint64_t`
-* `p` denotes a value of type `void const*`
-* `n` denotes a value of type `std::size_t`
-
-[table Hasher requirements
-[[operation] [type] [semantics, pre/post-conditions]]
-[
-    [`X a{s}`]
-    [ ]
-    [
-        `a` is constructed with a seed value integer. To achieve resistance
-        from algorithmic complexity attacks, an implementation of [*Hasher]
-        should ensure that values returned from the hash function will be
-        distinctly different for different values of `s` given the same
-        inputs. If algorithmic complexity attack resistance is not a
-        requirement, the seed may be ignored upon construction.
```

Comment:
I had to look back at the log to answer this.

I spent sometime to try to figure out how the hasher was used.  My conclusion was that it's usage was as a component of the algorithm which converts a key into an integer which can be used to select a bucket.  So the key requirement is that the hasher produce a uniform distribution of integers for the domain of the keys being used.  In this context, I didn't (and don't) understand what the cited sentence means. The phrase "algorithmic complexity attack" doesn't clarify anything for me.  If I've misunderstood what the hasher is for and how it's used we'll have to fix this up.  For the moment assume that I got it more or less right.

Basically I dropped the sentence because I didn't think it added anything (by itself anyway).